Crash when sharing to a tablet in landscape orientation

```
12-31 10:05:04.990    3165-3165/com.automattic.simplenote E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.automattic.simplenote, PID: 3165
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.automattic.simplenote/com.automattic.simplenote.NotesActivity}: java.lang.NullPointerException
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2195)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
            at android.app.ActivityThread.access$800(ActivityThread.java:135)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:136)
            at android.app.ActivityThread.main(ActivityThread.java:5017)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.NullPointerException
            at com.automattic.simplenote.NoteEditorFragment.setNote(NoteEditorFragment.java:225)
            at com.automattic.simplenote.NotesActivity.onNoteSelected(NotesActivity.java:652)
            at com.automattic.simplenote.NotesActivity.onCreate(NotesActivity.java:198)
            at android.app.Activity.performCreate(Activity.java:5231)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245)
            at android.app.ActivityThread.access$800(ActivityThread.java:135)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:136)
            at android.app.ActivityThread.main(ActivityThread.java:5017)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:515)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
            at dalvik.system.NativeStart.main(Native Method)
```

--- a/Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java
+++ b/Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java
@@ -179,11 +179,8 @@ public class NoteListFragment extends ListFragment implements AdapterView.OnItem
 	public void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
-		Simplenote application = (Simplenote) getActivity().getApplication();
-
 		mNotesAdapter = new NotesCursorAdapter(getActivity().getBaseContext(), null, 0);
 		setListAdapter(mNotesAdapter);
-
 	}
 
     @Override
@@ -472,15 +469,15 @@ public class NoteListFragment extends ListFragment implements AdapterView.OnItem
 
     public void setNoteSelected(String selectedNoteID) {
         // Loop through notes and set note selected if found
-        ObjectCursor<Note> cursor = (ObjectCursor<Note>)mNotesAdapter.getCursor();
-        if (cursor == null || cursor.getCount() == 0)
-            return;
-        for(int i=0; i < cursor.getCount(); i++) {
-            cursor.moveToPosition(i);
-            String noteKey = cursor.getSimperiumKey();
-            if (noteKey != null && noteKey.equals(selectedNoteID)) {
-                setActivatedPosition(i);
-                return;
+        ObjectCursor<Note> cursor = (ObjectCursor<Note>) mNotesAdapter.getCursor();
+        if (cursor != null) {
+            for (int i = 0; i < cursor.getCount(); i++) {
+                cursor.moveToPosition(i);
+                String noteKey = cursor.getSimperiumKey();
+                if (noteKey != null && noteKey.equals(selectedNoteID)) {
+                    setActivatedPosition(i);
+                    return;
+                }
             }
         }
 

--- a/Simplenote/src/main/java/com/automattic/simplenote/NotesActivity.java
+++ b/Simplenote/src/main/java/com/automattic/simplenote/NotesActivity.java
@@ -52,7 +52,7 @@ public class NotesActivity extends Activity implements
         NoteListFragment.Callbacks, User.StatusChangeListener, Simperium.OnUserCreatedListener, UndoBarController.UndoListener,
         Bucket.Listener<Note> {
 
-    private boolean mIsLargeScreen, mIsLandscape;
+    private boolean mIsLargeScreen, mIsLandscape, mShouldSelectNewNote;
     private String mTabletSearchQuery;
     private UndoBarController mUndoBarController;
     private SearchView mSearchView;
@@ -195,7 +195,8 @@ public class NotesActivity extends Activity implements
                 note.setModificationDate(note.getCreationDate());
                 note.setContent(text);
                 note.save();
-                onNoteSelected(note.getSimperiumKey(), true, null);
+                setCurrentNote(note);
+                mShouldSelectNewNote = true;
                 mTracker.sendEvent("note", "create_note", "external_share", null);
             }
         }
@@ -222,6 +223,11 @@ public class NotesActivity extends Activity implements
 
         updateNavigationDrawerItems();
         setSelectedTagActive();
+
+        if (mCurrentNote != null && mShouldSelectNewNote) {
+            onNoteSelected(mCurrentNote.getSimperiumKey(), true, null);
+            mShouldSelectNewNote = false;
+        }
     }
 
     @Override

